File: day15/reverse_sentence.py
def reverse_sentence():
    while True:
        user_input = input("\nPlease type I sentence and I will reverse it!\n")
        if not user_input.strip():
            print("\nPlease type at least one character.\n")
            continue

        return (user_input[::-1])
    
def print_reversed(user_string):
    print(user_string)

if __name__ == "__main__":
    user_sentence = reverse_sentence()
    print_reversed(user_sentence)

File: day15/reverse_sentence.rs
use std::io::{self, Write};

fn reverse_sentence() -> String {
    loop {
        print!("\nPlease type a sentence and I will reverse the word order:\n");
        io::stdout().flush().unwrap(); // ensure prompt appears before input

        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read input");

        let trimmed = input.trim();
        if trimmed.is_empty() {
            println!("\nPlease type at least one character.");
            continue;
        }

        let reversed: Vec<&str> = trimmed.split_whitespace().rev().collect();
        return reversed.join(" ");
    }
}

fn print_reversed(reversed: &str) {
    println!("{}", reversed);
}

fn main() {
    let result = reverse_sentence();
    print_reversed(&result);
}
